types/time.go

```diff
@@ -786,7 +786,8 @@ func isValidSeparator(c byte, prevParts int) bool {
 		return true
 	}
 
-	if prevParts == 2 && (c == ' ' || c == 'T') {
+	// add c == '\n' is for https://github.com/pingcap/tidb/issues/32232
+	if prevParts == 2 && (c == ' ' || c == 'T' || c == '\n') {
```

Comment:
According to MySQL [code](https://github.com/mysql/mysql-server/blob/8.0/mysys/my_time.cc#L121), it allow all the `space` char.

Reply:
done